src/cdi_ndi/NDI_Base.cc

```diff
@@ -23,26 +23,33 @@ namespace rtt_cdi_ndi {
  * For more details on NDI, see
  * https://xweb.lanl.gov/projects/data/nuclear/ndi/ndi.html
  *
- * \param[in] gendir_in path to gendir file
  * \param[in] dataset_in name of requested dataset (provided by inherited class)
  * \param[in] library_in name of requested NDI data library
  * \param[in] reaction_in name of requested reaction
- * \param[in] mg_form_in name of multigroup energy discretization to use
+ * \param[in] mg_e_bounds_in multigroup energy bin boundaries (keV)
  */
-NDI_Base::NDI_Base(const std::string &gendir_in, const std::string &dataset_in,
-                   const std::string &library_in,
-                   const std::string &reaction_in, const MG_FORM mg_form_in)
-    : gendir(gendir_in), dataset(dataset_in), library(library_in),
-      reaction(reaction_in), mg_form(mg_form_in) {
+NDI_Base::NDI_Base(const std::string &dataset_in, const std::string &library_in,
+                   const std::string &reaction_in,
+                   const std::vector<double> mg_e_bounds_in)
+    : dataset(dataset_in), library(library_in), reaction(reaction_in),
+      mg_e_bounds(mg_e_bounds_in) {
+  gendir = std::string(NDI_ROOT_DIR) + "share/gendir.all";
```

Comment:
1. move to initializer list
2. use `TO_NATIVE` to transform the filepath appropriately.
3. check that the file exists.

Reply:
To get the file existence check to work for all cases, I added a second constructor that takes a `gendir_in` argument and removed the `override_gendir_path` function from `NDI_Base`.